book.rb

```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_reader :rental
+  attr_accessor :title, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
+
+  def add_rental(person, date)
+    @rental.push(Rental.new(date, person, self)) unless @rental.include?(Rental.new(date, person, self))
+  end
```

Comment:
- [ ] this check will always end up pushing a new item. if you are creating a `Rental.new` it will never be anywhere else because you have just created it. you should just push the new book into the `@rental` array.